kuma/javascript/src/accountsettings/components/cancel-subscription-form.jsx

```diff
@@ -0,0 +1,78 @@
+// @flow
+import * as React from 'react';
+import { gettext, interpolate } from '../../l10n.js';
+import { GenericError } from '../../common/errors.jsx';
+import { deleteSubscriptions } from '../../payments/api.js';
+
+type Props = {
+    setShowForm: (((boolean) => boolean) | boolean) => void,
+    onSuccess: () => void,
+    date: string,
+};
+
+export const title = gettext('Are you sure you want to cancel?');
+
+const CancelSubscriptionForm = ({
+    setShowForm,
+    onSuccess,
+    date,
+}: Props): React.Node => {
+    const [status, setStatus] = React.useState<'error' | 'submitting' | 'idle'>(
+        'idle'
+    );
+
+    const handleCancel = () => setShowForm(false);
+
+    const handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        setStatus('submitting');
+
+        // success is handled by parent component
+        const handleSuccess = () => onSuccess();
+        const handleError = () => setStatus('error');
+        deleteSubscriptions(handleSuccess, handleError);
```

Comment:
I realize these API functions are older than this PR, but I still want to note that I think they should just use good-ol `return` and `throw` for success and failure, instead of callbacks. Do with that what you will 🙃

Reply:
I agree but, in the interest of time(I still need to implement close account and cycle back on user details) I am going to spin this of into a separate issue that I can cycle back to or, anyone can pick up if they have some spare cycles.

Reply:
https://github.com/mdn/kuma/issues/7286